atomic_reactor/plugins/pre_distribution_scope.py

```diff
@@ -0,0 +1,107 @@
+"""
+Copyright (c) 2016 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+from dockerfile_parse import DockerfileParser
+
+
+class NothingToCheck(Exception):
+    pass
+
+
+class DisallowedDistributionScope(Exception):
+    pass
+
+
+class DistributionScopePlugin(PreBuildPlugin):
+    """Apply distribution-scope rule.
+
+    Possible values for this label are:
+
+    (most restrictive)
+    - private
+    - authoritative-source-only
+    - restricted
+    - public
+    (least restrictive)
+
+    The rule we want to apply is to prevent images from having a
+    less restrictive scope than their parent images.
+
+    If the distribution-scope for this image is set to a less
+    restrictive value than the distribution-scope label inherited from
+    the parent image, fail the build.
+    """
+
+    # Name of this plugin
+    key = 'distribution_scope'
+
+    # The label and its possible values.
+    # https://github.com/projectatomic/ContainerApplicationGenericLabels
```

Comment:
Would it be better for this to be in the docstring where the values are listed out?